woocommerce/google-listings-and-ads
# Display Free Ad Credit notice only for new Google Ads sub account

### Changes proposed in this Pull Request:

Closes #340 . Depends on API created in #329.

This PR will check user's Google Ads account and display the free ad credit notice. The credit notice would only be shown if `sub_account` is `true` and `created_timestamp` is less than 30 days ago.

### Screenshots:

![image](https://user-images.githubusercontent.com/417342/112176174-d5506000-8c32-11eb-8a63-2621d0621d3a.png)

![image](https://user-images.githubusercontent.com/417342/112177175-ac7c9a80-8c33-11eb-8a27-678d6e763ef2.png)

### Detailed test instructions:

Pre-requisite (if you are not using a newly created Google Ads sub-account): 
1. Go to Connection Test, disconnect Ads account and create a new Ads account.
2. Check your email and accept invite. Note down the Google Ads account number.

Test - eligible for free ad credit:

1. Open https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fsetup-ads 
2. In Step 1, select the newly created Google Ads account based on the account number and connect the account.
3. Click Continue button to Step 2.
4. You should see the Free Ad Credit notice.
5. Select a target country and key in daily average cost. Make sure the UI is displayed ok.

Test - not eligible for free ad credit:

1. Go back to Connection Test. 
2. Disconnect Google Ads account.
3. Open https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fsetup-ads 
4. In Step 1, select the same disconnected Google Ads account and connect the account.
5. Click Continue button to Step 2.
6. You should _not_ see the Free Ad Credit notice.
7. Select a target country and key in daily average cost. Make sure the UI is displayed ok.

### Changelog Note:

> Check user's Google Ads account and determine whether they are eligible for free ad credit.
